packages/core/src/transport/WsOutboundTransport.ts

```diff
@@ -101,7 +102,13 @@ export class WsOutboundTransport implements OutboundTransport {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   private handleMessageEvent = (event: any) => {
     this.logger.trace('WebSocket message event received.', { url: event.target.url, data: event.data })
-    const payload = JSON.parse(Buffer.from(event.data).toString('utf-8'))
+    const responseMessage = Buffer.from(event.data).toString('utf-8')
```

Comment:
Not your pr, but `toString()` on a buffer is `UTF-8` by default.

Reply:
same here, you pass a string but don't parse the json in the isValidJweStructure body.